phoenix-core/src/main/java/org/apache/phoenix/iterate/TableResultIterator.java

```diff
@@ -143,6 +143,20 @@ public TableResultIterator(MutationState mutationState, Scan scan, ScanMetricsHo
         ScanUtil.setScanAttributesForClient(scan, table, plan.getContext().getConnection());
     }
 
+    public TableResultIterator(MutationState mutationState, Scan scan, ScanMetricsHolder scanMetricsHolder,
```

Comment:
Is this just visible for testing, or is this exposed to other things? (I believe this is a subtask of an improvement to hive-connector and spark-connector?)  

A Javadoc or comment explaining its purpose would be helpful

Reply:
@gjacoby126 @joshelser 
Actually I thought of adding the special constructor to use in phoenix spark connector to avoid fetching the metadata and avoid using the query plan but it's difficult to avoid using the query plan because we need the row projector, query context etc.. So I have found a simple solution of sharing the table meta data across the worker and cache it so that we need not fetch the metadata everytime in the workers and avoid the unnecessary load on RS holding the system catalog table. Here is the PR created under phoenix-connectors https://github.com/apache/phoenix-connectors/pull/80 with that changes in place this special constructor not required so will remove these special constructor changes and update the PR.